server/__test__/api/user.thisOrThatAnswers.js

```diff
@@ -0,0 +1,66 @@
+// testing get request for This Or That answers to display on profile
+
+const request = require("supertest");
+const mongoose = require("mongoose");
+
+// only need to build User data so doing a basic db connection
+const dbConnection = require("../../database/dbConnection");
+const resetDb = require("../../database/data/resetDB");
+
+const buildDatabase = require("../../database/data/development");
+
+const User = require("../../database/models/User");
+
+const app = require("../../app");
+
+// ROUTES
+const {
+  API_PROFILE_THISORTHAT_URL,
+  API_LOGIN_URL,
+} = require("../../../client/src/constants/apiRoutes");
+
+describe("Testing for profile-this-or-that route", () => {
+  beforeAll(async () => {
+    await dbConnection();
+    await resetDb();
+    await buildDatabase();
+  });
+
+  afterAll(async () => {
+    await resetDb();
+    mongoose.disconnect();
+  });
+
+  // get a logged in user you have in database
+  // log in with that user
+  // wrap your next call within the end of that one
+  test("test correct answer/question details", async (done) => {
+    const data = { email: "othman@shamla.com", password: "123456" };
+
+    request(app)
+      .post(API_LOGIN_URL)
+      .send(data)
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .end(async (err, result) => {
+        const token = result.headers["set-cookie"][0].split(";")[0];
+        const user = await User.find({ email: data.email });
+        const resultData = { userId: user[0]._id };
+
+        request(app)
+          .get(API_PROFILE_THISORTHAT_URL)
+          .set("Cookie", [token])
+          .send(resultData)
+          .end(async (error, res) => {
```

Comment:
We should check for the error first before completing the rest of the test
```suggestion
          .end(async (error, res) => {
             if (error) return done(error);
```